src/persist/src/indexed/cache.rs

```diff
@@ -137,8 +151,10 @@ impl<B: Blob> BlobCache<B> {
         let key = key.to_owned();
         // TODO: IO thread pool for persist instead of spawning one here.
         let _ = thread::spawn(move || {
+            let async_guard = cache.async_runtime.enter();
```

Comment:
Are the places where we spawn new threads the only places where this fixes real bugs, aside from tests where we use a `BlobCache` outside of a runtime context?